### ISSUE TITLE
run-examples fails on Linux

### ISSUE THREAD

**Issue by ronnieholm** [ChrisDill/Raylib-cs-Examples#14]
On Linux shouldn't the following steps just work?

    $ git clone https://github.com/ChrisDill/Raylib-cs-Examples.git
    $ make build
    $ make run-examples

Running the examples results in

```
Unhandled exception. System.DllNotFoundException: Unable to load shared library 'raylib' or one of its dependencies. In order to help diagnose loading problems, consider setting the LD_DEBUG environment variable: libraylib: cannot open shared object file: No such file or directory
   at Raylib_cs.Raylib.InitWindow(Int32 width, Int32 height, String title)
   at Examples.core_2d_camera_platformer.Main() in /tmp/Raylib-cs-Examples/Examples/core/core_2d_camera_platformer.cs:line 60
   at Examples.Program.RunCoreExamples() in /tmp/Raylib-cs-Examples/Examples/Program.cs:line 19
   at Examples.Program.Main(String[] args) in /tmp/Raylib-cs-Examples/Examples/Program.cs:line 8
Makefile:12: recipe for target 'run-examples' failed
make: *** [run-examples] Error 134
```

Am I missing anything obvious?

**Comment by ronnieholm** [comment]
> Absolutely! @ChrisDill and I both run Arch on our personal machines - which is why we are so confused.

That's really odd when you don't have raylib installed in one of the system paths.

I did come across others with a similar problem on Linux: https://github.com/dotnet/runtime/issues/34711

For now the way I resolve the issue is through the `ImportResolver` and `NativeLibrary.TryLoad`: https://stackoverflow.com/questions/58955723/net-core-project-with-native-library-depedency?noredirect=1&lq=1

**Comment by ronnieholm** [comment]
Good news. I was testing

    dotnet run --project Examples/Examples.csproj

on Ubuntu 20.04 and the examples load with both .NET 5 and .NET 6.

I then tried the same command on Ubuntu 18.04 and they still fail.

    LD_DEBUG=libs dotnet run --project Examples/Examples.csproj

As expected this results in `Unable to load shared library 'raylib' or one of its dependencies` and then later in the output:

    /lib/x86_64-linux-gnu/libm.so.6: error: version lookup error: version `GLIBC_2.29' not found (required by /home/rh/Downloads/raylibtest/Raylib-cs-Examples/Examples/bin/Debug/net6.0/runtimes/linux-x64/native/libraylib.so) (fatal)

Ubuntu 18.04 does't support GLIBC 2.29. Here's the version installed on Ubuntu 18.04 (Ubuntu 20.04 has 2.31).

    $ ldd --version ldd
    ldd (Ubuntu GLIBC 2.27-3ubuntu1.4) 2.27

Raylib-cs was last published on October 2, 2021. That is before this issue was created on October 24, 2021.

Odd thing is this glibc error didn't show up in the output last month. So I downgraded to .NET 5 on Ubuntu 18.04 and re-ran the `dotnet` command. The glibc error message still shows up. The message may originate from the native system loader (.NET may simply be passing it on) and not .NET. Perhaps the loader was updated sometime after October 24.

In conclusion: because libraylib.so is compiled against GLIBC 2.29, it cannot easily be consumed on Ubuntu 18.04.

**Comment by dsoronda** [comment]
Just a little info.
Examples works fine on Ubuntu 22.04.

**Comment by ronnieholm** [comment]
Resolving this issue. No sensible way to resolve on old versions of Ubuntu but works fine with recent Ubuntu.
